pkg/lib/bundle/generate.go

```diff
@@ -79,69 +70,84 @@ func GenerateFunc(directory, outputDir, packageName, channels, channelDefault st
 		}
 		log.Infof("Bundle %s information not provided, inferring from parent package directory",
 			strings.Join(notProvided, " and "))
 
-		i, err := NewBundleDirInterperter(directory)
+		i, err := NewBundleDirInterperter(bundleConfig.bundleDir)
 		if err != nil {
 			return fmt.Errorf("please manually input channels and packageName, "+
-				"error interpreting bundle from directory %s, %v", directory, err)
+				"error interpreting bundle from directory %s, %v", bundleConfig.bundleDir, err)
 		}
 
-		if channels == "" {
-			channels = strings.Join(i.GetBundleChannels(), ",")
-			if channels == "" {
+		if bundleConfig.channels == "" {
+			bundleConfig.channels = strings.Join(i.GetBundleChannels(), ",")
+			if bundleConfig.channels == "" {
 				return fmt.Errorf("error interpreting channels, please manually input channels instead")
 			}
-			log.Infof("Inferred channels: %s", channels)
+			log.Infof("Inferred channels: %s", bundleConfig.channels)
 		}
 
-		if packageName == "" {
-			packageName = i.GetPackageName()
-			log.Infof("Inferred package name: %s", packageName)
+		if bundleConfig.packageName == "" {
+			bundleConfig.packageName = i.GetPackageName()
+			log.Infof("Inferred package name: %s", bundleConfig.packageName)
 		}
 
-		if channelDefault == "" {
-			channelDefault = i.GetDefaultChannel()
-			if !containsString(strings.Split(channels, ","), channelDefault) {
-				channelDefault = ""
+		if bundleConfig.channelDefault == "" {
+			bundleConfig.channelDefault = i.GetDefaultChannel()
+			if !containsString(strings.Split(bundleConfig.channels, ","), bundleConfig.channelDefault) {
+				bundleConfig.channelDefault = ""
 			}
-			log.Infof("Inferred default channel: %s", channelDefault)
+			log.Infof("Inferred default channel: %s", bundleConfig.channelDefault)
 		}
 	}
 
 	log.Info("Building annotations.yaml")
 
 	// Generate annotations.yaml
-	content, err := GenerateAnnotations(mediaType, ManifestsDir, MetadataDir, packageName, channels, channelDefault)
+	content, err := GenerateAnnotations(Annotations{mediaType, ManifestsDir, MetadataDir, bundleConfig.packageName,
+		bundleConfig.channels, bundleConfig.channelDefault})
 	if err != nil {
 		return err
 	}
 
 	// Push the output yaml content to the correct directory and conditionally copy the manifest dir
-	outManifestDir, outMetadataDir, err := CopyYamlOutput(content, directory, outputDir, workingDir, overwrite)
+	outManifestDir, outMetadataDir, err := CopyYamlOutput(content, bundleConfig.bundleDir, bundleConfig.outputDir, bundleConfig.overwrite)
 	if err != nil {
 		return err
 	}
 
+	// Validate the output bundle
+	logger := log.NewEntry(log.New())
+	log.SetLevel(log.DebugLevel)
+
+	bundleValidator := NewBundleValidator(logger)
+	if err := bundleValidator.ValidateBundleFormat(bundleConfig.outputDir); err != nil {
+		return err
+	}
+
+	if err := bundleValidator.ValidateBundleContent(path.Join(bundleConfig.outputDir, ManifestsDir)); err != nil {
```

Comment:
right, didn't think about this. I am excluding validation in generate/build cmds. Since we are completely separating generate-csv feature with generate/build cmd, the cmds should resume it its original function to generate metadata folder with annotations and a dockerfile. This means:
- we are no longer changing bundle directory layout to bundle format having metadata, and manifests as subdirectories. This is achieveable given --output-dir flag.
- no longer validating the bundle in the generate/build cmds.